dreadnode/main.py

```diff
@@ -163,6 +179,121 @@ def _get_profile_api_key(self, profile: str | None = None) -> str | None:
         # Silently fail if profile config is not available or invalid
         return None
 
+    def _resolve_organization(self) -> None:
+        if self._api is None:
+            return
+
+        if self.organization:
+            self._organization = self._api.get_organization(self.organization)
+            if not self._organization:
+                raise RuntimeError(f"Organization '{self.organization}' not found.")
+
+        else:
+            organizations = self._api.list_organizations()
+
+            if not organizations:
+                raise RuntimeError(
+                    f"You are not part of any organizations on {self.server}. You will not be able to use Strikes.",
+                )
+
+            if len(organizations) > 1:
+                # We should not presume to choose an organization
+                org_list = "\t\n".join([f"- {o.name}" for o in organizations])
+                raise RuntimeError(
+                    f"You are part of multiple organizations. Please specify an organization from:\n{org_list}"
+                )
+            self._organization = organizations[0]
+
+    def _create_workspace(self, name: str) -> Workspace:
+        if self._api is None:
+            raise RuntimeError("API client is not initialized.")
+
+        try:
+            return self._api.create_workspace(name=name, organization_id=self._organization.id)
+        except RuntimeError as e:
+            if "403: Forbidden" in str(e):
+                raise RuntimeError(
+                    "You do not have permission to create the specified workspace for this organization. (You must be a Contributor to create new workspaces)."
+                ) from e
```

Comment:
Just checking a flow here, somebody tells a user to use "workspace-a", they put it in, what happens? If they can't access the workspace I assume we raise something other than a 404 on 229 below, but I want to confirm the assumption.

My worry is that a 404 is raised because the user can't "see" that workspace, and we try to create one.